tests/interfaces/records/handlers/records-query.spec.ts

```diff
@@ -112,6 +112,65 @@ describe('RecordsQueryHandler.handle()', () => {
       expect(reply2.entries?.length).to.equal(1); // only 1 entry should match the query
     });
 
+    it('should return records matching normalized protocol URI when querying protocol', async () => {
+      const alice = await TestDataGenerator.generatePersona();
+
+      // setting up a stub resolver
+      const mockResolution = TestDataGenerator.createDidResolutionResult(alice);;
+      sinon.stub(didResolver, 'resolve').resolves(mockResolution);
+
+      // install 3 different protocol URIs, two of which are the same after normalization
+      const protocol1 = 'tbd.website/dex';
+      const protocol2 = 'tbd.website/dex/';
+      const protocol3 = 'squareup.com/shop';
+      const protocols = [protocol1, protocol2, protocol3];
+      const protocolDefinition = emailProtocolDefinition; // To keep it simple, reuse the same definition for all 3 protocols
+      for (const protocol of protocols) {
+        const protocolConfigure = await TestDataGenerator.generateProtocolsConfigure({
+          requester: alice,
+          protocol,
+          protocolDefinition
+        });
+        const configureReply = await dwn.processMessage(alice.did, protocolConfigure.message);
+        expect(configureReply.status.code).to.equal(202);
+      };
+
+      // insert three messages into DB, two with matching protocol
+      for (const protocol of protocols) {
+        const write = await TestDataGenerator.generateRecordsWrite({
+          requester : alice,
+          protocol  : protocol,
+          schema    : protocolDefinition.labels.email.schema
+        });
+        const writeReply = await dwn.processMessage(alice.did, write.message, write.dataStream);
+        expect(writeReply.status.code).to.equal(202);
+      }
+
+      // Query for matching protocol
+      const messageData = await TestDataGenerator.generateRecordsQuery({
+        requester : alice,
+        filter    : {
+          protocol: protocol1
```

Comment:
should we also do a query for `protocol2` to make sure that it also matches the same entries?

Reply:
good idea. added